### PR TITLE
Add Lint ARM Templates step

### PR DESCRIPTION
**Pull request by romulets**
### Summary of your changes

Add ARM TTK Linter to the github workflow and fix linter errors

Test results available on https://github.com/elastic/cloudbeat/issues/1663#issuecomment-2072021094

### Related Issues
- Closes https://github.com/elastic/cloudbeat/issues/1663

### CODE REVIEW

**Review by kubasobon — APPROVED** [review]
LGTM when the templates are tested

File: .github/workflows/arm-template-lint.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+jobs:
+  lint-arm-ttk:
+    name: Lint
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        template: [ "ARM-for-organization-account.json", "ARM-for-single-account.json" ]
+    steps:
+      - uses: actions/checkout@v2
```
[the remark this answers is not in the document]
Comment (romulets) [reply .github/workflows/arm-template-lint.yml:22]: No, I'll update

**Review by kubasobon — APPROVED** [review]

File: deploy/azure/ARM-for-single-account.json [hunk 2]
```diff
@@ -217,29 +242,11 @@
...
-                            "properties": {
-                                "publisher": "Microsoft.ManagedIdentity",
-                                "type": "ManagedIdentityExtensionForLinux",
-                                "typeHandlerVersion": "1.0",
-                                "autoUpgradeMinorVersion": true,
-                                "settings": {
-                                    "port": 50342
-                                }
-                            }
-                        },
```
Comment (orestisfl) [inline deploy/azure/ARM-for-single-account.json:237]: Does this mean this is was not necessary before?

Comment (romulets) [reply deploy/azure/ARM-for-single-account.json:237]: It was. I discussed with @jeniawhite and it was there to grant access to the graph API.

The `Microsoft.Compute/virtualMachines/extensions` `Microsoft.ManagedIdentity` has been deprecated and should not be used. The only place I found a deprecation was on [azure cli release notes](https://learn.microsoft.com/en-us/cli/azure/release-notes-azure-cli#vm-27). I don't know the reason nor could find a documentation for a replacement. But still, that is a rule in the linter.

Through reading around (and some assumptions), it seems that the 

```
"identity": {
  "type": "SystemAssigned"
}
```

Should suffice nowadays giving a managed identity to the VM.

I tested and verified everything works as expected ([tests](https://github.com/elastic/cloudbeat/issues/1663#issuecomment-2074358752))

**Review by orestisfl — COMMENTED** [review]
(minor nitpick)

**Pull request merged**